clang/lib/Sema/SemaExpr.cpp

```diff
@@ -20694,6 +20720,9 @@ bool Sema::DiagnoseAssignmentResult(AssignConvertType ConvTy,
       AssignedVarFixIts;
   VarDecl *BoundsSafetyIncompatNestedPtrLocalVarToFix = nullptr;
   FixItHint BoundsSafetyIncompatNestedPtrFixIt;
+  bool InCXXBoundsSafetyMode = LangOpts.CPlusPlus &&
```

Comment:
Nit: This seems to be duplicated too in another Sema file. So it's another candidate to get a helper function.

Reply:
👌